File: array.c
#include <stdio.h>
#include "array.h"

void insertEnd(ArrayList* list, int value) {
    list->arr[list->size++] = value;
}

int getSizeArray(ArrayList* list) {
    return list->size;
}

void printArray(ArrayList* list) {
    for (int i = 0; i < list->size; i++) {
        printf("%d ", list->arr[i]);
    }
    printf("\n");
}

File: Array.java
public class Array {
    private int[] arr;
    private int size;

    public Array() {
        arr = new int[100];  // fixed capacity for assignment
        size = 0;
    }

    public void insertEnd(int value) {
        arr[size] = value;
        size++;
    }

    public int getSize() {
        return size;
    }

    public void printList() {
        for (int i = 0; i < size; i++) {
            System.out.print(arr[i] + "  ");
        }
        System.out.println();
    }
}